dashboard/models.py

```diff
@@ -46,6 +46,54 @@ class Meta:
         verbose_name = "Academic Terms"
         verbose_name_plural = "Academic Terms"
 
+
+class UserDefaultQuerySet(models.QuerySet):
+    def get_user_defaults(self, course_id, user_id, default_view_type):
+        try:
+            return self.get(course_id=str(course_id),
+                            user_id=str(user_id),
+                            default_view_type=str(default_view_type)).default_view_value
+        except (self.model.DoesNotExist, Exception) as e:
+            logger.error(f"""Couldn't get the default value for in course: {course_id} for user: {user_id} 
+                         with default_view_type: {default_view_type} due to {e} """)
+            return None
+
+    def set_user_default(self, course_id, user_id, default_view_type, default_view_value):
+        try:
+            return self.update_or_create(course_id=course_id, user_id=user_id, default_view_type=default_view_type,
+                                         defaults={'default_view_value': default_view_value})
+        except (self.model.DoesNotExist, Exception) as e:
+            logger.error(f"""Error when updating or creating default setting in course: {course_id} for user: {user_id} 
+                             with default_view_type: {default_view_type} and value: {default_view_value} due to {e} """)
+            raise e
+
+
+class UserDefaultManager(models.Manager):
+    def get_queryset(self):
+        return UserDefaultQuerySet(self.model, using=self._db)
+
+    def get_user_defaults(self, course_id, user_id, default_view_type):
+        return self.get_queryset().get_user_defaults(course_id, user_id, default_view_type)
+
+    def set_user_defaults(self, course_id, user_id, default_view_type, default_view_value):
+        return self.get_queryset().set_user_default(course_id, user_id, default_view_type, default_view_value)
+
+
+class UserDefaultSelection(models.Model):
+    id = models.AutoField(primary_key=True, verbose_name="Table Id")
+    user_id = models.CharField(max_length=255, verbose_name="SIS Id")
+    course_id = models.CharField(max_length=255, verbose_name="Course Id")
+    default_view_type = models.CharField(max_length=255, verbose_name="Default Type")
+    default_view_value = models.CharField(max_length=255, verbose_name="Default Value")
+
+    objects = UserDefaultManager()
+
+    class Meta:
+        managed = False
```

Comment:
Shouldn't this be managed = True if we're using the migrations?

Reply:
Ya. I wanted to know if we want to create tables using Django models or not for this release. I changed to True. We might want to enable all the table `managed = True` or make it a configurable option so that it can be turned on via .env